### ISSUE TITLE
False report of non-reactivity

### ISSUE THREAD

**Issue by CodeSmith32** [joshwilsonvu/eslint-plugin-solid#209]
**Describe the bug**
Reports that a prop reference inside JSX is not reactive.

**To Reproduce**
```tsx
import { createContext, type ParentProps } from "solid-js";

export const MyContext = createContext();

export const Provider = (props: ParentProps<{ value: string }>) => {
  // `props.value` here is used within JSX.
  // The warning says `'props.value' should be used within JSX`
  // ???
  return (
    <MyContext.Provider value={props.value}>
      {props.children}
    </MyContext.Provider>
  );
};
```

**Expected behavior**
No warning.

**Screenshots**

**Environment (please complete the following information):**

- OS: Windows 11
- Node version: v22.20.0
- `eslint-plugin-solid` version: 0.14.5
- `eslint` version: 9.39.2

**Additional context**

- [ ] I would be willing to contribute a PR to fix this issue

**Comment by CodeSmith32** [comment]
Apparently this is an intentional match, [according to the docs](https://github.com/solidjs-community/eslint-plugin-solid/blob/main/packages/eslint-plugin-solid/docs/reactivity.md). But I fail to see how this is correct behavior..?

**Comment by CodeSmith32** [comment]
I think I get why this reports a warning. Contexts pass down values as is. Therefore, passing a reactive function is generally required. But the rule doesn't recognize reactive functions, or objects containing a set of reactive functions, etc.
The problem is, reactivity for contexts is way too complex for the eslint plugin to accurately judge. Most of the time, this rule reports proper use of contexts as 'not reactive'.

I've turned this rule off (at least for now) because it just gets too annoying in too many places (even aside from contexts). I've learned the rules of reactivity. If I get bitten, I guess I'll suffer the consequence. This rule just seems to generate too many false positives to be useful for me... :-(

**Comment by ryansolid** [comment]
Confirmed on current main (0.16 + this week's fixes): `props.value` inside a `<MyContext.Provider value={...}>` attribute still incorrectly warns. Root cause looks like member-expression JSX element names not being recognized as component tracked scopes. Queued as a 0.16.1 false-positive fix.

**Comment by ryansolid** [comment]
Landed in v0.16.1, with a correction to my earlier diagnosis: the warning here is actually intentional. Providers read `value` once, untracked, when created — in Solid 1.x (`createRenderEffect` + `untrack`) and in 2.0 (`setContext(provider, props.value)` inside `createRoot`) — so `value={props.value}` genuinely freezes at the initial value. The real bug was the message: 'should be used within JSX' is nonsense for something that *is* in JSX. There's now a dedicated message explaining the freeze and suggesting you pass the signal/memo/store itself. Detection also covers the Solid 2.0 form where the context object is the provider (`<MyContext value={...}>`), resolved through `createContext()`.
